src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java

```diff
@@ -62,9 +66,12 @@ public List<Quiz> listRecentQuizzes() {
                 .toList();
     }
 
-    public Optional<QuizSimpleResult> updateQuizResult(UUID quizId, QuizSimpleResult quizSimpleResult) {
+    public QuizSimpleResult updateQuizResult(UUID quizId, QuizSimpleResult quizSimpleResult) {
+        if (findQuizEntity(quizId).isEmpty()) {
+            throw new ResourceNotFoundException(QUIZ);
+        }
         if (quizSimpleResult == null) {
```

Comment:
Is there any case when `quizSimpleResult` is null?

Reply:
yes, when quiz was just created and there is not any results yet